### ISSUE TITLE
Express Request with datatables server side not working postgres

### ISSUE THREAD

**Issue by sulistiana** [bhoriuchi/bookshelf-pagemaker#2]
please help me :(

controller:

```
pm(User, 'datatables').forge()
        .limit(req.query.start)
        .paginate({
            request: req,
            withRelated: ['location', 'division'],
        })
        .end()
        .then(function(results) {
                res.send(results);
        });
```

view:

```
$("#data-users").DataTable({
        "order": [[ 2, "asc" ]],
        "aoColumnDefs": [{ 'bSortable': false, 'aTargets': [0 , 1, 8] }],
        "processing": true,
        "serverSide": true,
        "ajax": {
            url: "/users/data",
            type: 'GET'
        },
    })
```

**Comment by bhoriuchi** [comment]
Sorry to hear that. Can you explain what specific error or undesired output you are getting.

For express please try using the `useQuery: true` option in paginate. This make paginate look in req.query instead of req.param for parameters.

Also I see in your controller code you are passing req.query.start to the limit method. Since you are already passing a request object to paginate, you can remove this method from the chain.

``` js
pm(User, 'datatables').forge()
        .limit(req.query.start)
        .paginate({
            useQuery: true,
            request: req,
            withRelated: ['location', 'division'],
        })
        .end()
        .then(function(results) {
                res.send(results);
        });
```

**Comment by bhoriuchi** [comment]
Also, I have published a new version `1.0.2` to npm that uses `request.query` by default so you may try the latest version and see if it works with your existing code.

**Comment by sulistiana** [comment]
yes it's work for request, but I found an error like this

![error](https://cloud.githubusercontent.com/assets/11223490/12548063/6a4a5542-c386-11e5-9b44-69a6e101a33a.png)

**Comment by bhoriuchi** [comment]
The debug shows you are using double quotes for the table name . SQL uses single quotes. Are you manually specifying this? Can you send me the full debug output

**Comment by sulistiana** [comment]
I can not determine their own, is directly generated by the bookshelf
this my code on server, database using postgre

```
pm(MdcLocation, 'datatables').forge()
        .paginate({
            request: req,
        })
        .end()
        .then(function(results) {

            res.send(results);
        });
```

**Comment by sulistiana** [comment]
error response

```
{"code":-1,"message":"An error was thrown during the paginate transaction","error":"42601","details"
:["select \"mdcLocations\".* from \"mdcLocations\" where 1 = 1 order by `mdcLocations`.`city` asc limit
 $1 - syntax error at or near \".\""],"stack":"error: select \"mdcLocations\".* from \"mdcLocations\"
 where 1 = 1 order by `mdcLocations`.`city` asc limit $1 - syntax error at or near \".\"\n    at Connection
.parseE (D:\\WORK\\DyCode\\danone-cms-pg\\node_modules\\pg\\lib\\connection.js:539:11)\n    at Connection
.parseMessage (D:\\WORK\\DyCode\\danone-cms-pg\\node_modules\\pg\\lib\\connection.js:366:17)\n    at
 Socket.<anonymous> (D:\\WORK\\DyCode\\danone-cms-pg\\node_modules\\pg\\lib\\connection.js:105:22)\n
    at emitOne (events.js:77:13)\n    at Socket.emit (events.js:169:7)\n    at readableAddChunk (_stream_readable
.js:146:16)\n    at Socket.Readable.push (_stream_readable.js:110:10)\n    at TCP.onread (net.js:523
:20)"}
```

**Comment by sulistiana** [comment]
its work for paginate if i change source code on bookshelf-pagemaker>lib>core>methods>paginate.js line 332

![error_pm](https://cloud.githubusercontent.com/assets/11223490/12576446/48caf21c-c446-11e5-86c3-1a694bdbf60c.png)

but still not working on search and order on datatables

**Comment by bhoriuchi** [comment]
please pass the `debug: true` option to the config hash for the sql debug output. it looks like the error is being caused by an empty value but ill need to see where exactly

example of config. i only really need the last couple commands before the exception

```js
var db = {
	"client": "pg",
	"connection": {
		"host": "127.0.0.1",
		"user": "db",
		"password": "password",
		"database": "test",
		"charset": "utf8"
	},
	"debug": true
};
```

**Comment by bhoriuchi** [comment]
i have found the issue. i do all my testing on mysql so i did not know that postgres does not understand &#x60;table&#x60;.&#x60;column&#x60;. I am removing the &#x60; and will post another version

**Comment by bhoriuchi** [comment]
v1.0.3 published to npm

**Comment by sulistiana** [comment]
still error on searching and ordering
query result when ordering request

```
select "mdcLocations".* from "mdcLocations" where 1 = 1 order by mdcLocations.city asc
```

i try on pg admin

```
select * from "mdcLocations" where 1 = 1 order by city asc
```

it's work

**Comment by bhoriuchi** [comment]
i have published a new version 1.0.4 to npm. after more testing and some extra code, the resolution for both search and order was to identify what type of database is being used and use the appropriate table/column escape character. for mysql/maria it is ` for postgres and all others it is "
